File: frontend/spacewise/src/app/models/blogpost.ts
import { User } from "./user"

export enum BlogpostCategory {
    CelestialBodies = 0,
    Physics = 1,
    Technology = 2,
    Other = 3
}

export interface BlogpostSection {
    id?: number,
    title?: string,
    content?: string
}

export interface Blogpost {
    id?: number,
    sections?: BlogpostSection[],
    category?: BlogpostCategory,
    image?: string,
    user?: User
}

File: backend/SpaceWise/SpaceWise/Models/Blogpost.cs
using SpaceWise.Models.Enums;

namespace SpaceWise.Models
{
    public class Blogpost
    {
        public required int Id { get; set; }
        public required ICollection<BlogpostSection> Sections { get; set; }
        public required BlogpostCategory Category { get; set; }
        public required string Image { get; set; }
        public required int UserId { get; set; }
        public required User User { get; set; }
    }
}
